server/src/fat/java/com/ibm/ws/lars/rest/FrontPageTest.java

```diff
@@ -58,18 +72,46 @@ public FrontPageTest(Protocol protocol, String url) {
 
     @Test
     public void testFrontPage() throws Exception {
-        HttpGet req = new HttpGet(baseUrl + "/");
-        String text = repository.doRequest(req, 200);
-        assertThat(text, containsString("The repository is running with 0 assets"));
+        JsonObject actualFpJsonObject = getFpJsonObject();
+
+        JsonObjectBuilder frontPageJsonBuilder = getJsonObjectBuilder();
+        frontPageJsonBuilder.add("serverName", "LARS");
+        frontPageJsonBuilder.add("assetCount", 0);
+        JsonObject expectedFrontPageJsonObject = frontPageJsonBuilder.build();
+
+        assertEquals("LARS front page JSON", expectedFrontPageJsonObject, actualFpJsonObject);
     }
 
     @Test
     public void testCorrectCount() throws Exception {
-        repository.addAssetNoAttachments(getTestAsset());
-        repository.addAssetNoAttachments(getTestAsset());
+        repository.addAssetNoAttachmentsWithState(getTestAsset(), State.PUBLISHED);
+        repository.addAssetNoAttachmentsWithState(getTestAsset(), State.AWAITING_APPROVAL);
+        repository.addAssetNoAttachmentsWithState(getTestAsset(), State.DRAFT);
+        repository.addAssetNoAttachmentsWithState(getTestAsset(), State.NEED_MORE_INFO);
+
+        JsonObject actualFpJsonObject = getFpJsonObject();
+        int actualAssetCount = actualFpJsonObject.getInt("assetCount");
+
+        assertEquals("LARS front page asset count", 1, actualAssetCount);
+    }
+
+    private JsonObject getFpJsonObject() throws ClientProtocolException, IOException {
         HttpGet req = new HttpGet(baseUrl + "/");
-        String text = repository.doRequest(req, 200);
-        assertThat(text, containsString("The repository is running with 2 assets"));
+        String fpText = repository.doRequest(req, 200);
+
+        InputStream fpInputStream = new ByteArrayInputStream(fpText.getBytes(StandardCharsets.UTF_8));
+        JsonReader fpJsonReader = Json.createReader(fpInputStream);
```

Comment:
Since you have already have the response as a String, you can just use a StringReader here, rather than converting the string to bytes and creating a ByteArrayInputStream.

Alternatively, we have a repository.doGetAsByteArray() method.